grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/GreedyTraversalPlan.java

```diff
@@ -257,6 +258,27 @@ private static void greedyTraversal(List<Fragment> plan, Arborescence<Node> arbo
         }
     }
 
+    private static double branchWeight(Node node, Arborescence<Node> arborescence,
+                                       Map<Node, Set<Node>> edgesParentToChild,
+                                       Map<Node, Map<Node, Fragment>> edgeFragmentChildToParent) {
+        final double[] weight = {getEdgeFragmentCost(node, arborescence, edgeFragmentChildToParent)};
+        weight[0] += nodeFragmentWeight(node);
+        if (edgesParentToChild.containsKey(node)) {
+            edgesParentToChild.get(node).forEach(child -> {
+                weight[0] += branchWeight(child, arborescence, edgesParentToChild, edgeFragmentChildToParent);
+            });
+        }
+        return weight[0];
+    }
+
+    private static double nodeFragmentWeight(Node node) {
+        return node.getFragmentsWithoutDependency().stream()
+                .map(Fragment::fragmentCost).reduce(0D, (x, y) -> x + y) +
+                node.getFragmentsWithDependencyVisited().stream()
+                        .map(Fragment::fragmentCost).reduce(0D, (x, y) -> x + y) +
+                node.getFixedFragmentCost();
```

Comment:
Streams have a method `count()` that does what you want here. Add some variables as well to make it a bit clearer :P

Reply:
I guess you mean method `sum()`, as fragment cost is not 1

Reply:
Oops! You're right. There is still a way to do it even neater:
`.mapToDouble(Fragment::fragmentCost).sum();`

Reply:
I think `sum()` only supports integer stream.